components/automate-cli/cmd/chef-automate/haNodeAdd_test.go

```diff
@@ -0,0 +1,372 @@
+	return msu.getClientConfigFunc()
+}
+func (msu *MockSSHUtilsImpl) getConnection() (*ssh.Client, error) {
+	return msu.getConnectionFunc()
+}
+func (msu *MockSSHUtilsImpl) connectAndExecuteCommandOnRemote(remoteCommands string, spinner bool) (string, error) {
+	return msu.connectAndExecuteCommandOnRemoteFunc(remoteCommands, spinner)
+}
+func (msu *MockSSHUtilsImpl) connectAndExecuteCommandOnRemoteSteamOutput(remoteCommands string) (string, error) {
+	return msu.connectAndExecuteCommandOnRemoteSteamOutputFunc(remoteCommands)
+}
+func (msu *MockSSHUtilsImpl) copyFileToRemote(srcFilePath string, destFileName string, removeFile bool) error {
+	return msu.copyFileToRemoteFunc(srcFilePath, destFileName, removeFile)
+}
+
+func Test_addnode_validate_error(t *testing.T) {
+	w := majorupgrade_utils.NewCustomWriterWithInputs("x")
+	flags := AddDeleteNodeHACmdFlags{
+		automateIp: "10.2.1.67,ewewedw",
+	}
+	nodeAdd := NewAddNode(w.CliWriter, flags, &MockNodeUtilsImpl{
+		readConfigfunc: func(path string) (ExistingInfraConfigToml, error) {
+			return readConfig(path)
+		},
+		getHaInfraDetailsfunc: func() (*SSHConfig, error) {
+			return &SSHConfig{}, nil
+		},
+	}, configtomlpath, &fileutils.MockFileSystemUtils{}, &MockSSHUtilsImpl{
+		connectAndExecuteCommandOnRemoteFunc: func(remoteCommands string, spinner bool) (string, error) {
+			return "", nil
+		},
+	})
+	err := nodeAdd.validate()
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "IP address validation failed: \nIncorrect Automate IP address format for ip ewewedw")
+}
+
+func Test_addnode_validate_error_multiple(t *testing.T) {
+	w := majorupgrade_utils.NewCustomWriterWithInputs("x")
+	flags := AddDeleteNodeHACmdFlags{
+		automateIp:   "10.2.1.67,ewewedw",
+		chefServerIp: "10.2.1.637,ewewedw",
+		postgresqlIp: "10.2.1.657,ewewedw",
+		opensearchIp: "10.2.1.61,ewewedw",
+	}
+	nodeAdd := NewAddNode(w.CliWriter, flags, &MockNodeUtilsImpl{
+		readConfigfunc: func(path string) (ExistingInfraConfigToml, error) {
+			return readConfig(path)
+		},
+		getHaInfraDetailsfunc: func() (*SSHConfig, error) {
+			return &SSHConfig{}, nil
+		},
+	}, configtomlpath, &fileutils.MockFileSystemUtils{}, &MockSSHUtilsImpl{
+		connectAndExecuteCommandOnRemoteFunc: func(remoteCommands string, spinner bool) (string, error) {
+			return "", nil
+		},
+	})
+	err := nodeAdd.validate()
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), `IP address validation failed: 
+Incorrect Automate IP address format for ip ewewedw
+Incorrect Chef-Server IP address format for ip 10.2.1.637
+Incorrect Chef-Server IP address format for ip ewewedw
+Incorrect OpenSearch IP address format for ip ewewedw
+Incorrect Postgresql IP address format for ip 10.2.1.657
+Incorrect Postgresql IP address format for ip ewewedw`)
+}
+
+func Test_addnode_readfile_error(t *testing.T) {
+	w := majorupgrade_utils.NewCustomWriterWithInputs("x")
+	flags := AddDeleteNodeHACmdFlags{
+		automateIp: "10.2.1.67",
```

Comment:
Replace all hardcoded IP addresses with one of the below suggestions:

- Loopback addresses 127.0.0.0/8 in CIDR notation (from 127.0.0.0 to 127.255.255.255)
- Broadcast address 255.255.255.255
- Non-routable address 0.0.0.0
- Strings of the form 2.5.<number>.<number> as they [often match Object Identifiers](http://www.oid-info.com/introduction.htm) (OID)
- Addresses in the ranges 192.0.2.0/24, 198.51.100.0/24, 203.0.113.0/24, reserved for documentation purposes by [RFC 5737](https://datatracker.ietf.org/doc/html/rfc5737)
- Addresses in the range 2001:db8::/32, reserved for documentation purposes by [RFC 3849](https://datatracker.ietf.org/doc/html/rfc3849)